### ISSUE TITLE
`grid` on dashboard resources doesn't allow multiple grids

### ISSUE THREAD

**Issue by tur-nr** [splunk-terraform/terraform-provider-signalfx#101]
We are trying to upgrade the provider to version 4 but all of our grids have broken 😢 

```terraform
resource "signalfx_dashboard" "detections" {
  name            = "Dashboard"
  dashboard_group = signalfx_dashboard_group.group.id

  // 2x1 grid (count=2)
  grid {
    chart_ids = ["1", "2"]
    width  = 6
  }
  
  // 3x3 grid (count=9)
  grid {
    chart_ids = ["3", "4", "5", "6", "7", "8", "9", "10", "11"]
    width  = 4
  }
}
```

What is the reason behind a single grid? As before this worked in version 3.

Related to #80

**Comment by cory-signalfx** [comment]
Hello @tur-nr. There's a reason, but it's not very interesting and doesn't do anything to fix your problems.

As I said in #80, I think I can fix this up. Thanks for the test case. I'll set it up locally and see what I can do.

**Comment by cory-signalfx** [comment]
Oops, this was closed in #102!
